furusiya/model/keys/callbacks.py

```diff
@@ -2,31 +2,45 @@
 from model.config import config
 
 
-def next_floor_callback():
-    Game.current_floor += 1
-
+def load_proper_area_map():
     Game.area_map.entities.remove(Game.player)
     if config.data.stallion.enabled:
         Game.area_map.entities.remove(Game.stallion)
 
     Game.area_map = Game.floors[Game.current_floor - 1]
 
-    Game.area_map.place_on_random_ground(Game.player)
+
+def switch_floor_callback(tile_to_spawn_player_around):
+    Game.area_map.place_around(Game.player, *tile_to_spawn_player_around)
```

Comment:
This code is really confusing. I understand how it works, but it's confusing; and having the stallion code in here makes it even more confusing.